satellite/console/service.go

```diff
@@ -229,30 +192,53 @@ func (s *Service) UpdateAccount(ctx context.Context, info UserInfo) (err error)
 	})
 }
 
-// ChangePassword updates password for a given user
-func (s *Service) ChangePassword(ctx context.Context, pass, newPass string) (err error) {
+// RequestChangePassword - sends email with link to restore forgotten password
+func (s *Service) RequestChangePassword(ctx context.Context, email string) (err error) {
 	defer mon.Task()(&ctx)(&err)
-	auth, err := GetAuth(ctx)
+
+	user, err := s.store.Users().GetByEmail(ctx, email)
 	if err != nil {
 		return err
 	}
 
-	err = bcrypt.CompareHashAndPassword(auth.User.PasswordHash, []byte(pass))
+	// TODO: use token variable instead of _
+	_, err = s.GenerateToken(ctx, user.ID, user.Email, user.CreatedAt.Add(tokenExpirationTime))
 	if err != nil {
-		return ErrUnauthorized.New("origin password is incorrect: %s", err.Error())
+		return err
 	}
 
-	if err := validatePassword(newPass); err != nil {
+	// TODO: wait till email service will be ready
+	// sendEmail(token)
+
+	return nil
+}
+
+// ChangePasswordExternal updates password for a given user in reset password flow
+func (s *Service) ChangePasswordExternal(ctx context.Context, resetPasswordToken, pass, newPass string) (err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	token, err := consoleauth.FromBase64URLString(resetPasswordToken)
+	if err != nil {
+		return
+	}
+
+	user, err := s.getUserFromToken(ctx, token)
```

Comment:
This should not skip checking token expiration time. Reset token can go through the same authenticate flow as authorization token. As we semantically defined it as separate self-encoded token type, may be should not dry out it's authorizing.